### ISSUE TITLE
Dynamic Template Mapping in `7.12.2` is missing

### ISSUE THREAD

**Issue by yannick-cw** [sksamuel/elastic4s#2437]
Hey, I am trying to upgrade from `7.12.0` to `7.12.2`.

I have some problems updating `DynamicTemplateRequest`

For example:

```scala
        val tsText = DynamicTemplateRequest("text_prefix_as_text", textField(""))
          .pathMatch("match_it")
```

Leads to an exception in elasticsearch:

```
["org.elasticsearch.index.mapper.MapperParsingException: Failed to parse mapping [_doc]: template must have mapping set"
```

The mapping field is missing on the dynamic template in the mapping request.

Seems to be this line in question:
https://github.com/sksamuel/elastic4s/blob/f3d242ccd4ca88b16b5a48ca7d9a98fa236edd32/elastic4s-handlers/src/main/scala/com/sksamuel/elastic4s/handlers/index/mapping/DynamicTemplateBodyFn.scala#L20

Is there a workaround or should that be uncommented again?

**Comment by jcostian-canter** [comment]
Discovered the same problem today, and prevents the update also for me. Hoping this gets fixed soon. Also it would be nice to see some migration guides, as for example the index creation and field mapping was changed quite much from 7.10 to 7.12.

**Comment by sksamuel** [comment]
Agreed. Much better docs planned for 8.0 if/when it's released :)

**Comment by jcostian-canter** [comment]
I was doing an update from 7.10 to 7.12(.2). It seems that the bigger API changes to the fields and dynamic mapping was done in 7.12.1 or 7.12.2.
7.12.0 seems to be working like before. I'm not sure if it's a good idea to introduce so big API changes in a point release.

**Comment by yannick-cw** [comment]
closed by #2430
